pkg/controllers/workapplier/controller_integration_test.go

```diff
@@ -3952,3 +3952,615 @@ var _ = Describe("report diff", func() {
 		})
 	})
 })
+
+var _ = Describe("switch apply strategies", func() {
+	Context("switch from report diff to CSA", Ordered, func() {
+		workName := fmt.Sprintf(workNameTemplate, "d1")
+		// The environment prepared by the envtest package does not support namespace
+		// deletion; each test case would use a new namespace.
+		nsName := fmt.Sprintf(nsNameTemplate, "d1")
```

Comment:
why not using the core name or a random suffix?

Reply:
Hi Ryan! Core-based suffix won't work because all of our integration tests are run using one process (`go test` rather than `ginkgo -p`); random suffix might make more sense, I could make the edit.